### ISSUE TITLE
[Feature request] Recursive find of projects in a folder

### ISSUE THREAD

**Issue by M-Gregoire** [bbatsov/projectile#1701]
I have a global projects folder which contains all my git repositories, but at different depths, similar to this:

```
projects/
├─ project1/
├─ group-of-project/
│  ├─ project2/
│  ├─ project3/
├─ go/
│  ├─ github.com/
│  │  ├─ username/
│  │  │  ├─ project4/

```

I'm trying to detect project 1 to 4 automatically with projectile (Issue related to https://github.com/bbatsov/projectile/issues/1500).

As advised in the [documentation](https://docs.projectile.mx/projectile/usage.html#automated-project-discovery), I've set 
```
(setq projectile-project-search-path '(("/path/to/projects" . 4)))
```

But I have a few issues:
- Only projects at **exactly** depth 4 are added, other projects are ignored
- A enormous amount of folders, that are not a project, are added, for examples, I have things such as:
```
...../project3/.git/objects/ff
...../project3/.git/objects/fe
...../project3/.git/objects/fd
...../project3/.git/objects/fc
...../project3/.git/objects/fb
...
```

Not specifiying a project search path and adding my projects manually does work, but is time consuming, especially as my number of projects grows.

Is there something I'm missing that would allow me to better handle my projects inside projectile without having to change my folder hierarchy ?

Thank you for any help you can provide

**Comment by bbatsov** [comment]
That's a tricky problem to solve and I haven't thought much about it, as I always keep all my projects at level 1 (usually I have one folder for personal project and one for work-related projects). Perhaps @juergenhoetzel or @mohkale will have more to say on the subject, as I recall they were active on #1500.

**Comment by juergenhoetzel** [comment]
>     * Only projects at **exactly** depth 4 are added, other projects are ignored

This is indeed the case (by design). 
I assumed that most users of this new feature use a tool like [ghq](https://github.com/x-motemen/ghq) that ensures a corresponding fixed directory level for the actual projects.

As a workaround you could use:
```elisp
(setq projectile-project-search-path '(("/path/to/projects" . 4)
				       ("/path/to/projects" . 3)
				       ("/path/to/projects" . 2)
				       ("/path/to/projects" . 1)))
```
>     * A enormous amount of folders, that are not a project, are added, for examples, I have things such as:
> 
> 
> ```
> ...../project3/.git/objects/ff
> ```

I can confirm this issue (fixed by https://github.com/bbatsov/projectile/pull/1704)

**Comment by M-Gregoire** [comment]
Thanks a lot for all those answers. I'm sorry I took so long to answer back!

With both examples from @mohkale and @juergenhoetzel , and with the fix from #1704 , my problem is now solved.

Thanks again to all participants.
